origin_ca.go

```diff
@@ -0,0 +1,153 @@
+package cloudflare
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+// OriginCA resource
+// This is the origin_ca resource definition (as defined here:
+// https://api.cloudflare.com/#origin-ca || https://api.cloudflare.com/#cloudflare-ca
+type OriginCA struct {
+	ID              string   `json:"id"`
+	Certificate     string   `json:"certificate"`
+	Hostnames       []string `json:"hostnames"`
+	ExpiresOn       string   `json:"expires_on"`
+	RequestType     string   `json:"request_type"`
+	RequestValidity string   `json:"requested_validity"`
+	Csr             string   `json:"csr"`
+}
+
+// OriginCertificateID is the resource sent back for a revoke action
+type OriginCertificateID struct {
+	ID string `json:"id"`
+}
+
+// OriginCAResponse is the APIv4 response envelop containing the OriginCA result
+type OriginCAResponse struct {
+	Success  bool     `json:"success"`
```

Comment:
Embed `Response` here to avoid repeating `Success` `Errors` and `Messages` each struct. Ref: https://github.com/cloudflare/cloudflare-go/blob/master/cloudflare.go#L152-L155